src/bin/mmix-sim.rs

```diff
@@ -0,0 +1,50 @@
+extern crate mmix;
+
+use mmix::machine::Machine;
+
+use std::fs::File;
+use std::io::{Error, Read};
+
+fn main() {
```

Comment:
It think, it is not.

I know that @jonas-schievink suggested this. But I think, that it is not the idiomatic way. As long as there is only one executable, it should be the default one.